packages/x-date-pickers-pro/src/DateRangePicker/DateRangePickerInput.tsx

```diff
@@ -144,7 +144,10 @@ export const DateRangePickerInput = React.forwardRef(function DateRangePickerInp
     lazyHandleChangeCallback([start, date], inputString);
   };
 
-  const openRangeStartSelection = () => {
+  const openRangeStartSelection = (
+    event: React.MouseEvent<HTMLElement> | React.KeyboardEvent<HTMLElement>,
```

Comment:
The following is enough if you just want to stop the propagation.
But if you think in the future we might use the event for other things, the more detailed typing can make sense

```suggestion
    event: React.UIEvent
```

Reply:
Yeah, I was debating about using a more common type, but IMHO—it seems a bit more logical to be explicit. Especially when we know what kind of events might be passed in. 🤔